src/model.ts

```diff
@@ -457,6 +457,19 @@ export class File {
   toString() {
     return "File [" + this.name + "]";
   }
+  isDescendantOf(element: File): boolean {
+    let parent = this.parent;
+    if (!parent) {
+      return false;
+    }
+    while (parent.parent) {
```

Comment:
I think this fails for `file.isDescendantOf(file.parent)`, if `file.parent.parent` is `null`.

Reply:
Yes you are right. Would this be a better approach?

```javascript
  isDescendantOf(element: File): boolean {
    let parent = this.parent;
    while (parent) {
      if (parent === element) {
        return true;
      }
      parent = parent.parent;
    }
    return false;
  }
```

Also, any pointers on where to create the test case? Can I create a new file called file.spec.tsx in tests/utils?

Reply:
You can probably just add a file to the `tests/unit` directory. Create a basic directory structure and check that `isDescendentOf` works correctly.